draft-ietf-wimse-client-assertion-in-workload-environments.md

```diff
@@ -0,0 +1,546 @@
+
+A solution to this problem is to not provision secret material to the workload and use the platform the workload runs on to attest for it. Many workload platforms offer a credential, in most cases a JWT token. Signed by a platform-internal authorization server, the credential attests the workload and its attributes. Based on {{RFC7521}} and its JWT profile {{RFC7523}}, this credential can then be used as a client assertion towards a different authorization server.
+
+# Terminology
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and
+"OPTIONAL" in this document are to be interpreted as described in
+BCP 14 {{RFC2119}} {{RFC8174}} when, and only when, they appear in all
+capitals, as shown here.
+
+For the scope of this specification, the term authorization server is only used for the authorization server of the external authorization domain as highlighted in {{fig-overview}}.
+
+Even though, technically, the platform credential is also issued by an authorization server within the workload platform, this specification only refers to it as "platform issuer" or just "platform".
+
+# OAuth Assertion in Workload Environments
+
+## Overview
+
+{{fig-overview}} illustrates a generic pattern that applies across all of the patterns described in {{patterns}}:
+
+~~~ aasvg
+ +----------------------------------------------------------+
+ |                           External Authorization Domain  |
+ |                                                          |
+ | +-------------------------+     +--------------------+   |
+ | |                         |     |                    |   |
+ | |   Authorization Server  |     | Protected Resource |   |
+ | |                         |     |                    |   |
+ | +-------^---------+-------+     +----------^---------+   |
+ +---------+---------+------------------------+-------------+
+           |         |                        |
+           |   3) access token           4) access
+           |         |                        |
+2) present assertion |                        |
+           |         |     +------------------+
+           |         |     |
+ +---------+---------+-----+--------------------------------+
+ |         |         |     |             Workload Platform  |
+ |         |         |     |                                |
+ |  +------+---------v-----+---+           +-------------+  |
+ |  |                          |           | Credential  |  |
+ |  |        Workload          +-----------> issued by   |  |
```

Comment:
```suggestion
 |  |        Workload          +<---------> issued by   |  |
```

Reply:
@hannestschofenig this edit misaligns the diagram, was the intent to just make it a double-headed arrow? If so that syntax would be as follows:

```suggestion
 |  |        Workload          <-----------> issued by   |  |
```

Reply:
Latest commit makes the arrow going both ways.